src/components/kubectl/kubeconfig.ts

```diff
@@ -2,21 +2,59 @@ import * as vscode from 'vscode';
 import { fs } from '../../fs';
 import * as path from 'path';
 import * as yaml from 'js-yaml';
+import * as shelljs from 'shelljs';
 import { refreshExplorer } from '../clusterprovider/common/explorer';
-import { getActiveKubeconfig } from '../config/config';
+import { getActiveKubeconfig, getUseWsl } from '../config/config';
 
 interface Named {
     readonly name: string;
 }
 
+export function getKubeconfigPath(): { pathType: 'host'; hostPath: string } | { pathType: 'wsl'; wslPath: string }  {
+    // If the user specified a kubeconfig path -WSL or not-, let's use it.
+    let kubeconfigPath: string | undefined = getActiveKubeconfig();
+
+    if (getUseWsl()) {
+        if (!kubeconfigPath) {
+            // User is using WSL: we want to use the same default that kubectl uses on Linux ($KUBECONFIG or home directory).
+            const result = shelljs.exec('wsl.exe sh -c "${KUBECONFIG:-$HOME/.kube/config}"', { silent: true }) as shelljs.ExecOutputReturnValue;
+            if (!result) {
+                throw new Error(`Impossible to retrieve the kubeconfig path from WSL. No result from the shelljs.exe call.`);
+            }
+
+            if (result.code !== 0) {
+                throw new Error(`Impossible to retrieve the kubeconfig path from WSL. Error code: ${result.code}. Error output: ${result.stderr.trim()}`);
+            }
+            kubeconfigPath = result.stdout.trim();
+        }
+        return {
+            pathType: 'wsl',
+            wslPath: kubeconfigPath
+        };
+    }
+
+    if (!kubeconfigPath) {
+        kubeconfigPath = process.env['KUBECONFIG'];
+    }
+    if (!kubeconfigPath) {
+        // Fall back on the default kubeconfig value.
+        kubeconfigPath = path.join((process.env['HOME'] || process.env['USERPROFILE'] || '.'), ".kube", "config");
+    }
+    return {
+        pathType: 'host',
+        hostPath: kubeconfigPath
+    };
+}
+
 export async function mergeToKubeconfig(newConfigText: string): Promise<void> {
-    const kcfile = kubeconfigPath();
-    if (!(await fs.existsAsync(kcfile))) {
-        vscode.window.showErrorMessage("Couldn't find kubeconfig file to merge into");
+    const config = getKubeconfigPath();
```

Comment:
>config [](start = 10, length = 6)

Nit: we have this pattern multiple times, where we first call getKubeconfigPath and then reads its result the next line.
Because "config" is an ambiguous name, I'd suggest to go for "kubeconfigPath" instead. I agree that it's similar to the other "kubeconfigFilePath" variable declared, but I think that's OK, and looks better than using "config" in my opinion.